db.py

```diff
@@ -211,10 +240,12 @@ def query(
         result = db.OFFSET(result, offset)
     if limit:
         result = db.LIMIT(result, limit)
+    if distinct:
```

Comment:
This should go in a separate commit and needs separate tests. Additionally, in the SQL order of execution, I think it goes between select and order by

Reply:
(test is slightly less important but it should go in a separate commit than type checking for sure)